parser/parser/include/ruby_parser/driver.hh

```diff
@@ -164,11 +190,12 @@ public:
 // The stack has 3 states:
 //  * `top = 0`: no parameter (ordinary or numbered) in this scope
 //  * `top < 0`: ordinary parameter(s) in this scope
-//  * `top > 0`: at leat one numbered parameter in this scope (top being the highest one found)
+//  * `top > 0`: at least one numbered parameter in this scope (top being the highest one found)
 class max_numparam_stack {
     struct NumparamScope {
         int max;
         ruby_parser::node_list *decls;
+        bool staticContext = false;
```

Comment:
`staticContext` is used while iterating over the numparam stack. This follows from the whitequark implementation https://github.com/sorbet/sorbet/pull/5725/files#diff-b6ead23655c4ab14d663a4d1ad737bbf8634556bb7e2890b71d1606260c4b101R202